tests/baselines/reference/useBeforeDeclaration_destructuring.types

```diff
@@ -1,19 +1,19 @@
 === tests/cases/compiler/useBeforeDeclaration_destructuring.ts ===
 a;
->a : any
+>a : string
 
 let {a, b = a} = {a: '', b: 1};
->a : any
->b : any
->a : any
+>a : string
+>b : string | number

```

Comment:
I feel like this probably could be improved as it's impossible for `b` to be of type `string` here. Perhaps this doesn't have to be improved in this PR though and a followup issue could be created to track this.

Reply:
Totally agree.